src/screencast/pipewire_screencast.c

```diff
@@ -166,27 +166,32 @@ void xdpw_pwr_stream_init(struct xdpw_screencast_instance *cast) {
 	enum spa_video_format format = xdpw_format_pw_from_wl_shm(cast);
 	enum spa_video_format format_without_alpha =
 		xdpw_format_pw_strip_alpha(format);
-	uint32_t n_formats = 1;
+
+	struct spa_pod_frame f;
+	spa_pod_builder_push_object(&b, &f, SPA_TYPE_OBJECT_Format, SPA_PARAM_EnumFormat);
+	spa_pod_builder_add(&b, SPA_FORMAT_mediaType, SPA_POD_Id(SPA_MEDIA_TYPE_video), 0);
+	spa_pod_builder_add(&b, SPA_FORMAT_mediaSubtype, SPA_POD_Id(SPA_MEDIA_SUBTYPE_raw), 0);
 	if (format_without_alpha != SPA_VIDEO_FORMAT_UNKNOWN) {
-		n_formats++;
+		spa_pod_builder_add(&b, SPA_FORMAT_VIDEO_format,
+			SPA_POD_CHOICE_ENUM_Id(3, format, format, format_without_alpha), 0);
+	} else {
+		spa_pod_builder_add(&b, SPA_FORMAT_VIDEO_format,
+			SPA_POD_CHOICE_ENUM_Id(2, format, format), 0);
 	}
-
-	const struct spa_pod *param = spa_pod_builder_add_object(&b,
-		SPA_TYPE_OBJECT_Format, SPA_PARAM_EnumFormat,
-		SPA_FORMAT_mediaType,       SPA_POD_Id(SPA_MEDIA_TYPE_video),
-		SPA_FORMAT_mediaSubtype,    SPA_POD_Id(SPA_MEDIA_SUBTYPE_raw),
-		SPA_FORMAT_VIDEO_format,    SPA_POD_CHOICE_ENUM_Id(n_formats + 1,
-			format, format, format_without_alpha),
-		SPA_FORMAT_VIDEO_size,      SPA_POD_CHOICE_RANGE_Rectangle(
+	spa_pod_builder_add(&b, SPA_FORMAT_VIDEO_size,
+		SPA_POD_CHOICE_RANGE_Rectangle(
 			&SPA_RECTANGLE(cast->simple_frame.width, cast->simple_frame.height),
 			&SPA_RECTANGLE(1, 1),
 			&SPA_RECTANGLE(4096, 4096)),
-		// variable framerate
-		SPA_FORMAT_VIDEO_framerate, SPA_POD_Fraction(&SPA_FRACTION(0, 1)),
-		SPA_FORMAT_VIDEO_maxFramerate, SPA_POD_CHOICE_RANGE_Fraction(
+		0);
+	spa_pod_builder_add(&b, SPA_FORMAT_VIDEO_framerate, SPA_POD_Fraction(&SPA_FRACTION(0, 1)), 0);
```

Comment:
This adds a `0`. Is this something we were missing too?

Reply:
Oh, no, just an arg for `spa_pod_builder_add`.

Reply:
I simply expanded the `spa_pod_builder_add_object` macro that was been used before, so no difference. You can see the macro definition in `/usr/include/spa-0.2/spa/pod/builder.h`